### PR TITLE
Adding Markdown Table Reporting to Summary, and more ....

### PR DESCRIPTION
**Pull request by aaronlippold**
- [x] Refactored and simplified the summary code
- [x] Documented all class, methods and objects
- [x] Added a more logical set of flags so that you could both output to stdout and a file at the same time or either
- [x] Added Winston Logging
- [x] Added Unit Tests for Summary
	- [x] Markdown
	- [x] YAML
	- [x] JSON 
- [x] Updated README documentation
- [x] Added TSDoc Documentation to all updated and added files and classes
	- [x] (not this pr) Need to add generation and publication as we move work into ts-ohdf and others
- [x] Add usage examples
- [x] Find language and a markdown formatter that looks good in README (hint: its `powershell` and pick your words well)
- [x] Started building the OHDF Types and moving things into the start of a module
	- [x] `utils/ohdf/types.ts` 
	- [x] `utils/ohdf/dataloader.ts`
	- [x] `utils/ohdf/outputGenerator.ts`
	- [x] `utils/ohdf/calculations.ts`
- [x] Added tests for all the utility functions added
	- [x] OHDF Dataloader
	- [x] OHDF Calculations
	- [x] OHDF Generator
- [x] Added developer docs
- [x] Fixed `npm run dev` and documented in Dev docs

Signed-off-by: Aaron Lippold <lippold@gmail.com>

### CODE REVIEW

**Review by Amndeep7 — COMMENTED** [review]
Look over the winston changes.  I think Aaron said he changed the default to print to file unless you change some env flag.  I don't mind either way but we should make sure to document somewhere what that flag ought to be and also make sure the CI has that flag set.

File: new-summary.ts
Comment (aaronlippold) [reply new-summary.ts]: just testing and plaseholder code

File: README.md
Comment (em-c-rod) [inline README.md:1063]:
```suggestion
Generate a comprehensive summary of compliance data, including totals and counts, from your OHDF files.
```

File: README.md
Comment (em-c-rod) [inline README.md:1067]: I didn't catch it - what is the difference in display for the `shell` or `powershell` code blocks? That way I know when to use which

File: src/commands/view/summary.ts
Comment (em-c-rod) [inline src/commands/view/summary.ts:130]:
```suggestion
      this.logger.verbose(`Loaded execution data from ${this.parsedFlags.input.length.toString()} file(s)`)
```

Copilot suggests this because it would otherwise print [object Object] supposedly.

File: src/commands/view/summary.ts
Comment (em-c-rod) [inline src/commands/view/summary.ts:134]:
```suggestion
      this.logger.verbose(`Calculated total counts for ${summaries.length.toString()} summaries`)
```

File: src/commands/view/summary.ts
Comment (em-c-rod) [inline src/commands/view/summary.ts:136]:
```suggestion
      this.logger.verbose(`Calculated compliance scores for ${executionData.length.toString()} execution data`)
```

File: src/commands/view/summary.ts
Comment (em-c-rod) [inline src/commands/view/summary.ts:141]:
```suggestion
      this.logger.verbose(`Generated ${printableSummaries.length.toString()} printable summaries`)
```

File: src/commands/view/summary.ts
Comment (em-c-rod) [reply src/commands/view/summary.ts:130]: But I didn't check in the logging if this is needed.

File: src/commands/view/summary.ts
Comment (em-c-rod) [inline src/commands/view/summary.ts:78]:
```suggestion
    help: Flags.help({char: 'h', description: 'Show help information', helpGroup: 'help'}),
```

Consider adding this so the help output doesn't just say "FLAGS" and require the user to see that there are more types of flags underneath.

File: src/utils/logging.ts
Comment (em-c-rod) [inline src/utils/logging.ts:33]: Where does this get stored? I agree with @Amndeep7 that if we want to change the default, we should make sure to document where it's going and where to find it. Maybe we leave the default as outputting to the console and update Winston logging in an overall refactor ticket after this PR?

File: src/utils/ohdf/calculations.ts
Comment (em-c-rod) [inline src/utils/ohdf/calculations.ts:6]: Does it make sense to add these to this file instead of having them in the threshold util? It seems we want to have overall utils organized based on the functionality rather than where they are used and it makes sense to me to have the calculate stuff together.

File: src/commands/view/summary.ts
Comment (em-c-rod) [inline src/commands/view/summary.ts:63]: Sometimes I see this is `view summary:` and sometimes it is `View Summary`. Not a huge deal, but we could make sure all of the winston logger instances are consistent or add this to a winston logger refactor ticket if it is decided to move in that direction.

File: README.md
Comment (aaronlippold) [reply README.md:1067]: Just did this update in the Summary example for now. 

To not 'bloom' the PR too much, I will add that change to the rest after we merge this change.

File: src/commands/view/summary.ts
Comment (aaronlippold) [reply src/commands/view/summary.ts:130]: Please go ahead and make this addition

File: src/commands/view/summary.ts
Comment (aaronlippold) [reply src/commands/view/summary.ts:132]: Same as above, please make this addition.

File: src/commands/view/summary.ts
Comment (aaronlippold) [reply src/commands/view/summary.ts:134]: see above....

File: src/commands/view/summary.ts
Comment (aaronlippold) [reply src/commands/view/summary.ts:136]: :) ☝🏻

File: src/commands/view/summary.ts
Comment (aaronlippold) [reply src/commands/view/summary.ts:141]: ☝🏻

File: src/utils/logging.ts
Comment (aaronlippold) [reply src/utils/logging.ts:33]: This will be 'fixed' when we do the logging PR. At the moment this is just a bit of cruft. I will see if I can work on it tonight to move this around but if not, we can merge and address it directly after.

File: src/utils/ohdf/calculations.ts
Comment (aaronlippold) [reply src/utils/ohdf/calculations.ts:6]: I didn't write that code, but will be rewriting it as part of the thresholds PR.

Short answer:

No given I think this calculations parent will have more kids soon ...

Longer Answer:

Given the complexity or hopefully not so complex after, we will likely have a construction like:

```shell
./src/utils/ohdf/calculations/thresholds.ts
...../same/summary.ts
.../same/other thing we have to count.ts
```

I am finding, as with the outputGenerators, which I make make this change, that scoping things to their utilityTYpe is really the way to go.

File: src/commands/view/summary.ts
Comment (aaronlippold) [reply src/commands/view/summary.ts:63]: This will be generalized and come from the class name in a PR near you is my guess so we can leave it for then

File: src/utils/logging.ts
Comment (aaronlippold) [reply src/utils/logging.ts:33]: technically it goes in the $PWD of where you exec'd the command which isn't correct - it should - default to either the defined $SAF_CLI_LOG env var first, then to `~/.saf-cli/` second in a big boy application.

We can add that as a point on the logger PR

File: src/utils/ohdf/calculations.ts
Comment (aaronlippold) [reply src/utils/ohdf/calculations.ts:6]:
```suggestion
import {calculateCompliance, extractStatusCounts, renameStatusName, severityTargetsObject} from '../threshold'
```

Create an issue we can attach to the PR for 'fixing summary'

File: src/commands/view/summary.ts
Comment (aaronlippold) [reply src/commands/view/summary.ts]: Plese go ahead with that change and we can see if we want to standardize that across the commands, I think in the end we will as part of the general cleanup of user doc

File: README.md
Comment (aaronlippold) [reply README.md]: Seems to be fixed already.

File: docs/contributors-guide.md
Comment (em-c-rod) [inline docs/contributors-guide.md:7]:
```suggestion
- Node.js (version 18.18 or newer)
```

File: docs/contributors-guide.md
Comment (em-c-rod) [inline docs/contributors-guide.md:7]:
```suggestion
- Node.js (check the package.json file for the current version)
```

File: run-script-os
Comment (wdower) [inline run-script-os:1]: was this intentionally committed @aaronlippold ?

File: run-script-os
Comment (em-c-rod) [inline run-script-os:1]: Is this file intentional? @aaronlippold

**Review by wdower — CHANGES_REQUESTED** [review]
LGTM except wondering if that run-script-os is supposed to be here

**Review by em-c-rod — APPROVED** [review]
Same comment as Will on the run-script-os. I tried some incorrect files and paths and the errors seemed intuitive.

**Review by em-c-rod — CHANGES_REQUESTED** [review]
Same comment as Will on the run-script-os. I tested with various input paths and the errors seemed intuitive

File: run-script-os
Comment (aaronlippold) [reply run-script-os:1]: yes the thats managed by another plugin, it works, since your `npm run test` works.

File: run-script-os
Comment (Amndeep7) [reply run-script-os:1]: i'm pretty sure it shouldn't be there.  run-script-os is the utility we're using to do cross-platform stuff.  i don't think there should be a file called that that contains a couple random unicode characters.

Comment (aaronlippold) [reply run-script-os:1]: Yes its a binary file by one of the utility npm packages for identifying which os your system is so it knows which version of command:win32 or command:linux or ... will run. Its not a script - its a compiled binary I think which is why it looks odd when you open it.

**Comment by aaronlippold** [comment]
> Same comment as Will on the run-script-os. I tested with various input paths and the errors seemed intuitive

This is not a script its a binary so you are not suposed to run it yourself, npm run ... does

**Comment by Amndeep7** [comment]
> > Same comment as Will on the run-script-os. I tested with various input paths and the errors seemed intuitive
> 
> This is not a script its a binary so you are not suposed to run it yourself, npm run ... does

Yeah and the binary is in node_modules.  Why is it at the top level instead now.  And also i'm pretty sure the actual binary is a lot larger than 7 bytes.

File: run-script-os
Comment (aaronlippold) [reply run-script-os]: Found it, Fixed it.

File: src/utils/ohdf/outputGenerator.ts
Comment (Amndeep7) [inline src/utils/ohdf/outputGenerator.ts:208]: could probably return the expression outright here instead of assigning to the 'tables' variable.

File: src/utils/ohdf/outputGenerator.ts
Comment (Amndeep7) [inline src/utils/ohdf/outputGenerator.ts:286]:
```suggestion
  const profile_name = typeof item.profileName === 'string' ? item.profileName : 'Evaluated Profile'
```

**Review by Amndeep7 — CHANGES_REQUESTED** [review]

File: README.md
Comment (aaronlippold) [reply README.md:14]: Fixed

File: docs/contributors-guide.md
Comment (aaronlippold) [reply docs/contributors-guide.md:1]: already fixed

File: docs/contributors-guide.md
Comment (aaronlippold) [reply docs/contributors-guide.md:3]: Fixed

File: .eslintrc
Comment (aaronlippold) [reply .eslintrc:24]: This may be OBE given some of the updates to the tests, I can double check.

File: docs/contributors-guide.md
Comment (aaronlippold) [reply docs/contributors-guide.md:14]: Powershell will be the standard one to use

File: docs/contributors-guide.md
Comment (aaronlippold) [reply docs/contributors-guide.md:25]: You are suggesting we spell it all out?

File: package.json
Comment (aaronlippold) [reply package.json:190]: fixed

File: package.json
Comment (aaronlippold) [reply package.json:191]: you mean its location in the scripts hash?

File: src/utils/logging.ts
Comment (aaronlippold) [reply src/utils/logging.ts:33]: Issue opened for this. We will open a PR for logging in the next itterations.

**Pull request merged**
